### PR TITLE
Create new movie page w/ form error handling

### PR DESCRIPTION
**Pull request by liishi**
- updated API end point for create action
- built movie form container and error list components
- added POST#create movie controller test
- updated config/routes for create action

### CODE REVIEW

File: app/javascript/react/components/MoviesFormContainer.js [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        headers: {
+          'Accept': 'application/json',
+          'Content-Type': 'application/json'
+        }
+      })
+      .then(response => {
+        if (response.ok) {
+          return response.json()
+        } else {
+          // error stuff
```
Comment (dovakeen118) [inline app/javascript/react/components/MoviesFormContainer.js]: We need _real_ error stuff here with a catch statement!

File: spec/controllers/api/v1/movies_controller_spec.rb [hunk 2]
```diff
@@ -34,18 +34,33 @@
...
+      post :create, params: { movie: {title: "Terminator", summary: "Arnold's voice", year: 1984} }
+
+      returned_json = JSON.parse(response.body)
+      expect(response.status).to eq 200
+      expect(response.content_type).to eq("application/json")
+      expect(returned_json).to be_kind_of(Hash)
+      expect(returned_json).to_not be_kind_of(Array)
+      expect(returned_json["title"]).to eq "Terminator"
+      expect(returned_json["summary"]).to eq "Arnold's voice"
+      expect(returned_json["year"]).to eq 1984
```
Comment (dovakeen118) [inline spec/controllers/api/v1/movies_controller_spec.rb]: We should also test a failed POST request

**Review by dovakeen118 — CHANGES_REQUESTED** [review]
Need to include a failed POST request and error handling for form submission in the fetch request

File: app/javascript/react/components/MoviesFormContainer.js (hunk 1 above)
Comment (liishi) [reply app/javascript/react/components/MoviesFormContainer.js]: added!

File: spec/controllers/api/v1/movies_controller_spec.rb (hunk 2 above)
Comment (liishi) [reply spec/controllers/api/v1/movies_controller_spec.rb]: added! additionally, added tests to see if the object persists in the database.

File: app/javascript/react/components/ErrorList.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+      <div className="callout alert">
+        <ul>{listItems}</ul>
+      </div>
+    )
+  } else {
+    return ""
+  }
+}
+
+export default ErrorList
```
Comment (safang99) [inline app/javascript/react/components/ErrorList.js:26]: Very nice!

**Pull request merged**